Assets/Mirror/Runtime/Messages.cs

```diff
@@ -13,6 +13,14 @@ public abstract class MessageBase
 
         // Serialize the contents of this message into the writer
         public virtual void Serialize(NetworkWriter writer) {}
+
+        public static int GetId<T>() where T: MessageBase
+        {
+            // paul: 16 bits is enough to avoid collisions
+            //  - keeps the message size small because it gets varinted
+            //  - in case of collisions,  Mirror will display an error
+            return typeof(T).FullName.GetStableHashCode() & 0xFFFF;
```

Comment:
why not return ushort like before?
not sure if it's worth preparing for the day when we need 'int', since a lot of the time we would only save 1 byte of bandwidth because of how writepacked works

Reply:
Because you wanted it to be configurable in one place.   If we want 32 bits instead of 16 bits all you have to do is change it here.

If I we make this method a short,  now you have to change the type all over the place.

Reply:
do you still prefer ushort?   I can easily do it,  just throwing away one of the commits.

Reply:
it's fine, your reasoning makes sense